specification/flag-evaluation/flag-evaluation.md

```diff
@@ -0,0 +1,188 @@
+
+#### API Initialization and Configuration
+
+##### Requirement 1.1
+
+> The `API`, and any state it maintains **SHOULD** exist as a global singleton, even in cases wherein multiple versions of the `API` are present at runtime.
+
+It's important that multiple instances of the `API` not be active, so that state stored therein, such as the registered `provider`, static global `evaluation context`, and globally configured `hooks` allow the `API` to behave predictably. This can be difficult in some runtimes or languages, but implementors should make their best effort to ensure that only a single instance of the `API` is used.
+
+##### Requirement 1.2
+
+> The `API` **MUST** provide a function to set the global `provider` singleton, which accepts an API-conformant `provider` implementation.
+
+```
+// example provider mutator
+OpenFeature.setProvider(new MyProvider());
+```
+
+See [provider](../provider//providers.md) for details.
+
+##### Requirement 1.3
+
+> The `API` **MUST** provide a function to add `hooks` which accepts one or more API-conformant `hooks`, and appends them to the collection of any previously added hooks. When new hooks are added, previously added hooks are not removed.
+
+```
+// example hook attachment
+OpenFeature.addHooks([new MyHook()]);
+```
+
+See [hooks](./hooks.md) for details.
+
+##### Requirement 1.4
+
+> The API **MUST** provide a function for retrieving the `provider` implementation.
+
+```
+// example provider accessor
+OpenFeature.getProvider();
+```
+
+See [provider](../provider/provider.md) for details.
+
+##### Requirement 1.5
+
+> The `API` **MUST** provide a function for creating a `client` which accepts the following options:
+>
+> - name (optional): A logical string identifier for the client.
+
+```
+// example client creation and retrieval
+OpenFeature.getClient({
+  name: 'my-openfeature-client'
+});
+```
+
+The name is a logical identifier for the client.
+
+#### Client Usage
+
+##### Requirement 1.6
+
+> The client **MUST** provide a method to add `hooks` which accepts one or more API-conformant `hooks`, and appends them to the collection of any previously added hooks. When new hooks are added, previously added hooks are not removed.
+
+```
+// example hook attachment
+client.addHooks([new MyHook()]);
+```
+
+See [hooks](./hooks.md) for details.
+
+#### Flag Evaluation
+
+##### Requirement 1.7
+
+> The `client` **MUST** provide methods for flag evaluation, with parameters `flag key` (string, required), `default value` (boolean | number | string | structure, required), `evaluation context` (optional), and `evaluation options` (optional).
+
+```
+// example flag evaluation
+var myValue = client.getValue('my-flag', false);
+```
+
+##### Condition 1.8
+
+> The language type system differentiates between strings, numbers, booleans and structures.
+>
+> ##### Conditional Requirement 1.8.1
```

Comment:
I still don't think we want these in blockquotes.

Reply:
I agree, but for now our fancy little W3C-QA-compliant spec parser requires it. I will make an issue to resolve that. I agree it looks ugly.

Added chore issue: https://github.com/open-feature/spec/issues/37